src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,127 @@
+package duke.parser;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.commands.*;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.ui.Ui;
+
+/**
+ * Represents a parser to parse inputs from the user.
+ */
+public class Parser {
+    private static DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+    private static DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HHmm");
+    private static DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+
+    /**
+     * Makes sense of the user input and returns the
+     * corresponding command to be executed.
+     *
+     * @param input Input from the user.
+     * @param ui Ui to interact with the user.
+     * @return Command to be executed.
+     */
+    public static Command parse(String input, Ui ui) {
+        String[] arr = input.split(" ", 2);
+        String command = arr[0];
+
+        switch (command) {
+        case ByeCommand.COMMAND_WORD:
+            return new ByeCommand();
+        case ListCommand.COMMAND_WORD:
+            return new ListCommand();
+        case MarkCommand.COMMAND_WORD:
+            try {
+                return new MarkCommand(Integer.parseInt(arr[1]) - 1);
+            } catch (NumberFormatException | IndexOutOfBoundsException e) {
+                ui.invalidMarkInput();
+            }
+            return null;
+        case UnmarkCommand.COMMAND_WORD:
+            try {
+                return new UnmarkCommand(Integer.parseInt(arr[1]) - 1);
+            } catch (NumberFormatException | IndexOutOfBoundsException e) {
+                ui.invalidUnmarkInput();
+            }
```

Comment:
I like that you use switch cases for numerous conditions, it looks much cleaner. Great work!